ports/stm32/mpthreadport.h

```diff
@@ -32,7 +32,9 @@ typedef pyb_mutex_t mp_thread_mutex_t;
 void mp_thread_init(void);
 void mp_thread_gc_others(void);
 
-static inline void mp_thread_set_state(void *state) {
+struct _mp_state_thread_t;
```

Comment:
Any way to do it without this forward declaration? Perhaps to put it somewhere common and not port-specific?

Reply:
This is an odd case because everything is defined as a static inline function and this file is included in py/mpthread.h which has the same declaration of `struct _mp_state_thread_t` already

So the alternative would be to move the declaration in that file before the `#include` of this file.

Reply:
Either that or to place the first declaration in another file, included by both (like a `types.h` file). But by doing your suggestion, you make `mpthreadport.h` un-includable from other files that don't include `mpthread.h` (I dunno if that happens?). If not - I guess it's okay to move it.

Reply:
> So the alternative would be to move the declaration in that file before the `#include` of this file.

I think this is a good option.  Since `mpthreadport.h` is like a configuration file for `py/mpthread.h` the former should not really be included on its own, only via the latter (similar to how `mpconfigport.h` is not included directly).

Well, esp32 seems to include `mpthreadport.h` directly but it should still work because it doesn't refer to `_mp_state_thread_t` (and it can be just left as-is).

Summary: please shift the declaration up in `py/mpthread.h` so it doesn't need to be added to `ports/stm32/mpthreadport.h`.

Reply:
I have updated it with the suggested change.